### ISSUE TITLE
Resolving merge conflicts externally has weird UX

### ISSUE THREAD

**Issue by peppy** [jesseduffield/lazygit#1325]
Two weird UX scenarios around the resolution of merge conflicts when using an external means of resolution:

Firstly, the resolution is not detected by lazygit until the view all files which have outstanding conflicts. Even using the refresh key doesn't seem to help:

https://user-images.githubusercontent.com/191335/118959719-8a866680-b99d-11eb-8872-5fc2782b87e2.mp4

I remember mentioning this in another issue thread and you said that conflicting files would appear at the top of the tree view (which would make "visiting" them easier, as is the workaround for this issue), but that doesn't seem to be the case.

Secondly, the merge is completely aborted if you select all files in the staging area:

https://user-images.githubusercontent.com/191335/118959971-c4576d00-b99d-11eb-9354-ba19dd9575e3.mp4

**Comment by peppy** [comment]
To bump this with another use case which I regularly encounter: I've been keeping several branches up-to-date with a rapidly changing `master`. Usually this involves fixing 3 or 4 conflicts amongst hundreds of files changed. Locating the files with conflicts from within lazygit is not a simple task:



https://user-images.githubusercontent.com/191335/121465866-60025900-c9f1-11eb-924b-72507a83c1d6.mp4

**Comment by jesseduffield** [comment]
@peppy sorry that I haven't gotten around to tackling this on yet. As for viewing files with merge conflicts, you can switch to the flat view with the backtick key which has merge conflict files sorted to the top by default

**Comment by jesseduffield** [comment]
@peppy I've spent some time today looking at a fix for https://github.com/jesseduffield/lazygit/issues/1718 and came across this issue in the process. It turns out that if you fix merge conflicts in your editor and save that file, git doesn't actually treat the files as resolved until you `git add` them. At the moment, when going to render a file in the main view that supposedly has merge conflicts, if it turns out it doesn't actually have merge conflicts, we stage the file for you and then check to see if everything is merged.

So there's two questions on my mind:
1) are there situations where a user would _not_ want lazygit to stage these files on their behalf, upon discovering that the merge conflicts are all resolved (I haven't heard anybody complain about it so I'm guessing no)
2) is there a cheap way of checking whether all files with merge conflicts are now resolved?

If the logic is that whenever there's files with merge conflicts and we've just refreshed the files view we scan each file for conflict markers to see if we can stage them, I wonder if that would be expensive when somebody's dealing with large files and/or many files. Then again it might not be expensive at all, I haven't actually measured it (and will do that right now haha).
